Allow module to use the general "markup" config

MikaYuoadas opened the pull request:
Not sure if that's the right way to do this, but here's what I use for my setup.

In lots of my module config `format` string I use pango markup.
But it isn't interpreted (and instead display the ugly xml in my i3bar) unless I use the following patch.

Here's an example extracted from my setup that needs this to work and properly display the font awesome icons:
```
general {
    color_good = "#3EC2FF"
    color_bad = "#FF0000"
    colors = true
    output_format = "i3bar"
    markup = "pango"
}

order += "volume_status"

volume_status {
    format = "<span font='Font Awesome 5 Free Solid'></span> {percentage}%"
    format_muted = "<span font='Font Awesome 5 Free Solid'></span>"
    color_muted = "#FFFF00"
}
```

ultrabug commented:
Hi @MikaYuoadas thanks a lot for your contribution.

I've not looked into it yet but will ASAP, just saying thanks and acknowledging that we'll check it out :+1:

tobes commented:
Your patch is pretty good but I'd do this then we can do `markup` in the `general`/`py3status` blocks or in just the module we want

```diff
diff --git a/py3status/module.py b/py3status/module.py
index 290d8b2c..20b4dc5f 100644
--- a/py3status/module.py
+++ b/py3status/module.py
@@ -350,6 +350,11 @@ class Module:
 
             self.i3s_module_options['separator_block_width'] = sep_block_width
 
+        fn = self._py3_wrapper.get_config_attribute
+        param = fn(self.module_full_name, 'markup')
+        if not hasattr(param, 'none_setting'):
+            self.i3s_module_options['markup'] = param
+
     def process_composite(self, response):
         """
         Process a composite response.
```

ultrabug commented:
@MikaYuoadas could you update to reflect on @tobes comment please?

lasers commented:
This patch could check first for `pango`, `none` too.

tobes commented:
Sure @lasers no harm in that

lasers commented:
@tobes Merge this as-is? I bring out my PR next.

MikaYuoadas commented:
Sorry for the delay, I've been pretty busy lately.

I've rebased the branch on latest `master` and changed the code to reflect @tobes' comment.
I'm not sure what @lasers about checking `pango`, `none` meant though.

tobes commented:
@lasers comment was about checking the value is pango or none.  but I don't think it's very important.

MikaYuoadas commented:
[i3bar's code](https://github.com/i3/i3/blob/next/i3bar/src/child.c#L222) currently only check if the value is "pango" and ignore any other value.
Maybe we shouldn't check for any particular value in case in the future i3bar supports more markup options?

ultrabug commented:
Thanks a lot @MikaYuoadas !
